tests/framework/extensions/clusters/clusters.go

```diff
@@ -74,6 +75,19 @@ func IsProvisioningClusterReady(event watch.Event) (ready bool, err error) {
 	return ready && updated, nil
 }
 
+// IsProvisioningClusterReadyAndUpdated is basic check function that would be used for checking if the cluster is ready after an update.
+func IsProvisioningClusterReadyAndUpdated(cluster *apisV1.Cluster) (ready bool, err error) {
```

Comment:
Earlier we were using KubeProvisioningClient which has a "Watch" method and which is currently being used in the other tests as well. 
Since Steve client doesn't have "Watch" method, I could not make this function to support Watch interface, hence used the poll method.

Reply:
I totally get your point. As you mentioned, `KubeProvisioningClient` is still being used for other tests. As you explained the reason, it is fine to instantiate the dynamic client with `KubeProvisioningClient` but _only_ to watch a resource. This could be used in your tests too. (Small note: it's always better to instantiate the dynamic client and watch interface from the admin framework client to prevent any RBAC issue.)

I made this comment cause the watch logic and watched parameters are the same as the above extension `IsProvisioningClusterReady`. Let's say this was watching a v1 cluster's Kubernetes version, then I wouldn't have any questions on that since we don't have any standard to watch (can be poll or watch, it's the test writer's preference).

Alternatively, If you think this extension is necessary for other writers, changing the name might make this clearer, as this is the same watch above. Maybe `IsProvisioningClusterReadyWithPoll`?

Reply:
So Rishabh, the watch is a Kubernetes specific functionality, the Steve client wouldn't be able to do that. That being said it is totally ok to use the watch interface to wait for resources. So in this case I agree with Ugur, use the method for waiting that is totally ok.

Reply:
Okay, so changed the function name to `IsProvisioningClusterReadyWithPoll` so that it can be used with steve client.

Reply:
Rishab I think this is a redundant function.  Let me be more clear, it's ok to use the KubeProvisioningClient for watching resources created by Steve, we do this in our provisioning tests.  For watching/waiting we can use the KubeProvisioningClient, but for things that can be done in the UI, we strictly use Steve. Make sense?

Reply:
Okay, I've removed the `IsProvisioningClusterReadyWithPoll` function and have used KubeProvisioningClient now.